Homebrew 3.2.0 (#665)
Homebrew/brew.sh: _posts/2021-06-21-homebrew-3.2.0.md

---
title: 3.2.0
author: MikeMcQuaid
redirect_from: /blog/3.2.0/
---
Today I'd like to announce Homebrew 3.2.0. The most significant changes since 3.1.0 are `brew install` now upgrades outdated formulae by default and basic macOS 12 (Monterey) support.

Major changes and deprecations since 3.1.0:

> - [`brew install` now upgrades outdated formulae by default (for indempotency). This can be disabled by setting [`HOMEBREW_NO_INSTALL_UPGRADE`].](https://github.com/Homebrew/brew/pull/11525)

Reviewer: ```suggestion
- [`brew install` now upgrades outdated formulae by default (for idempotency). This can be disabled by setting `HOMEBREW_NO_INSTALL_UPGRADE`.](https://github.com/Homebrew/brew/pull/11525)
```

Typo, and the extra square brackets looked weird.

- [`brew` has basic macOS 12 (Monterey) support.](https://github.com/Homebrew/brew/pull/11504)
- [`brew leaves` has `--installed-on-request` and `--installed-as-dependency` flags to only list formula installed manually or as dependencies.](https://github.com/Homebrew/brew/pull/11368)
- [Various deprecations, disables and code deletions for Homebrew 3.2.0.](https://github.com/Homebrew/brew/pull/11548)

Other changes since 3.1.0 I'd like to highlight are the following:

- [`brew link --HEAD` links the `HEAD` version of a formula.](https://github.com/Homebrew/brew/pull/11397)
- [`brew alias`, `brew autoupdate` and `brew command-not-found` are official external command taps.](https://github.com/Homebrew/brew/pull/11137)
- [`brew tap` will never create shallow clones (as shallow clones are not performant when repeatedly fetching as Homebrew does).](https://github.com/Homebrew/brew/pull/11337)
- [`brew fetch` will no longer use shallow clones.](https://github.com/Homebrew/brew/pull/11328)
- [`brew install` also outputs cask caveats as part of the final summary.](https://github.com/Homebrew/brew/pull/11509)
- [`brew` has GCC 11 support.](https://github.com/Homebrew/brew/pull/11299)

> - [`brew bottle` will generate `all:` bottles which are used on all platforms.](https://github.com/Homebrew/brew/pull/11215). A lot of work has gone into improving [reproducible builds](https://reproducible-builds.org) so these are possible. `brew bottle` on your local machine for an `all:` bottle should generate an identical checksum.

Reviewer 2: ```suggestion
- [`brew bottle` will generate `all:` bottles which are used on all platforms.](https://github.com/Homebrew/brew/pull/11215). A lot of work has gone into improving [reproducible builds](https://reproducible-builds.org) to make these possible. `brew bottle` on your local machine for an `all:` bottle should generate an identical checksum.
```

- [`brew doctor` hides some warnings on Apple Silicon with Intel and ARM installations.](https://github.com/Homebrew/brew/pull/11540)
- [`brew bottle --bottle-arch` allow bottles with custom architectures.](https://github.com/Homebrew/brew/pull/11352)

Finally:

- [Homebrew accepts donations through GitHub Sponsors](https://github.com/sponsors/Homebrew) and [still accepts donations through Patreon](https://www.patreon.com/homebrew). If you can afford it, please consider donating. If you'd rather not use GitHub Sponsors or Patreon (our preferred donation methods), [check out the other ways to donate in our README](https://github.com/homebrew/brew/#donations).

> Thanks to all our hard-working maintainers, contributors, sponsors and supporters for getting us this far. Particular thanks on Homebrew 3.1.0 go to GitHub for providing GitHub Packages and helping in our migration. Enjoy using Homebrew!

Reviewer: ```suggestion
Thanks to all our hard-working maintainers, contributors, sponsors and supporters for getting us this far.
```
